spec/system/banners/new_spec.rb

```diff
@@ -49,14 +49,29 @@
     within "#banners" do
       click_on "Edit", match: :first
     end
-    fill_in "banner_expires_at", with: 7.days.ago.strftime("%m%d%Y\t%I%M%P")
+    fill_in "banner_expires_at", with: 2.days.from_now.strftime("%m%d%Y\t%I%M%P")
     click_on "Submit"
 
     visit banners_path
     expect(page).to have_text("Expiring Announcement")
 
     visit root_path
-    expect(page).not_to have_text("Please fill out this survey.")
+    expect(page).to have_text("Please fill out this survey.")
+  end
+
+  it "does not allow creation of banner with an expiration time set in the past" do
+    sign_in admin
+
+    visit banners_path
+    click_on "New Banner"
+    fill_in "Name", with: "Annoucement"
+    fill_in "banner_expires_at", with: 1.hour.ago.strftime("%m%d%Y\t%I%M%P")
+    fill_in_rich_text_area "banner_content", with: "Please fill out this survey."
+    click_on "Submit"
+
+    message = page.find("#banner_expires_at").native.attribute("validationMessage")
+    expect(message).to start_with("Value must be")
+    # Can't get an exact match for the date/time since cookie "browser_time_zone" isn't set in the test Chrome session
```

Comment:
This test fails for me locally. In my case the browser is providing a cookie with the EST timezone but the test and server are assuming UTC. So `1.hour.ago` gets converted from EST to UTC but upon input it gets treated as it if was EST since that is what the cookie says to it end up not actually being an hour in the past.

Basically if its 1AM now => 1 hour ago = 12AM => inputted in UTC = 5AM (that is what goes into the field)
But the browser has a cookie set for EST so it converts that to UTC = 10AM UTC which is after 6AM (1 AM EST / 6AM UTC) so there is no error.

Long story short: `page.driver.browser.manage.add_cookie(name: "browser_time_zone", value: "Something")` to set a cookie.

Reply:
Thank you! I changed the test to specify a timezone and it's working for me locally. Let me know if it still fails on your end.

Reply:
The test now fails in CI though. I think that the browser in CI is not picking up the timezone cookie because this is the assertion that fails (for test running at 5:08pm CST):

```
expected: "Value must be 06/11/2024, 05:08 PM or later."
            got: "Value must be 06/11/2024, 10:08 PM or later."
```

The time we are getting is in UTC, although CST is requested [here](https://github.com/rubyforgood/casa/blob/c0164bddf43d32ad26816636abc7319a6dee8477/spec/system/banners/new_spec.rb#L76).